components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -254,6 +261,14 @@ void BravePrefProvider::UpdateCookieRules(ContentSettingsType content_type,
     }
   }
 
+  if (prefs_->GetBoolean(kGoogleLoginControlType)) {
```

Comment:
not sure if this overrides user shields settings or not.
Anyway, this method becomes too fat, so i'd prefer to put it into a separate method (or at least add a comment).

Reply:
it shouldn't override shields settings as long as it comes last, but it should have a test to verify

Reply:
Test added here: https://github.com/brave/brave-core/pull/3784/files#diff-a3b1f2bffde11cef192a185a65955508R350

Reply:
although actually we want it to override site shield settings because it should work the same whether the default is block 3p cookies or whether the default is something else and site shield settings are block 3p cookies. In the case where site settings are block all you won't be able to login to anything anyway